pipelines/common/src/main/java/org/openmrs/analytics/JdbcConnectionPools.java

```diff
@@ -103,6 +103,9 @@ private static DataSource createNewPool(
     // Lowering the minimum pool size to limit the number of connections if multiple pools are
     // created for the same DB.
     comboPooledDataSource.setMinPoolSize(1);
+    // Reducing acquire retry attempts value so that the clients waiting on the connections fail
+    // sooner, but keeping it sufficient to increase the successful connection.
+    comboPooledDataSource.setAcquireRetryAttempts(10);
```

Comment:
I still don't think this is the right approach for testing if the input DB config is correct. We should not need to wait 100 seconds to know this. Here is my suggestion: When validating configuration parameters, e.g., [here](https://github.com/google/fhir-data-pipes/blob/3694f1394d0b9011ab480ca61f0bd0568bca2f53/pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java#L102), create a temporary direct Connection (_not_ through the connection pool) and fail early if the connection cannot be established. There seems to be this driver specific way to see if the connection [isValid](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#isValid-int-) which we can use too.

Reply:
BTW, if you make the change I have suggested above, please revert this `setAcquireRetryAttempts` change.

Reply:
In the latest commit, below changes are made
1. On application launch, the configurations parameters of jdbc and FHIR server are validated depending on the mode that has been enabled.
2. Changes to `setAcquireRetryAttempts` have been reverted.